build.rs

```diff
@@ -547,10 +548,24 @@ fn obj_path(out_dir: &Path, src: &Path) -> PathBuf {
 
 fn configure_cc(c: &mut cc::Build, target: &Target, c_root_dir: &Path, include_dir: &Path) {
     let compiler = c.get_compiler();
-    // FIXME: On Windows AArch64 we currently must use Clang to compile C code
-    let compiler = if target.os == WINDOWS && target.arch == AARCH64 && !compiler.is_like_clang() {
-        let _ = c.compiler("clang");
-        c.get_compiler()
+    // FIXME: On Windows AArch64, ring C code cannot be compiled using cl.exe, but must be compiled
+    // using the LLVM toolchain.
+    let compiler = if target.os == WINDOWS && target.arch == AARCH64 {
+        match compiler.family() {
+            // If the user is already using clang or clang-cl, nothing needs to be replaced
+            cc::ToolFamily::Clang { .. } | cc::ToolFamily::Msvc { clang_cl: true, .. } => compiler,
```

Comment:
I suggest we deleted this case. Either we detect non-clang-cl MSVC and then switch it to clang-cl because we know cl.exe doesn't work, or we do nothing.

Reply:
Not sure I understand, this already does nothing. Unless you simply mean that when replacing the below panic with `_ => compiler,`, this is a duplicate match arm and would intrinsically be removed anyway?